### DISCUSSION TITLE
Why wrapping around the `git` command instead of C API?

### DISCUSSION THREAD

**Discussion by dclong** [gitpython-developers/GitPython#1258] [Q&A] [answered]
GitPython is essentially a wrapper around the `git` command. Why doesn't it call the corresponding C APIs?

**Comment by Byron** [comment, marked as answer]
Back in 2008 at its inception there was no libgit2 or there were no libgit2 bindings I presume, so it started out wrapping the `git` command.
When I started getting into it I worked with what's there, leveraging the `git` program wherever possible.

For many `GitPython` is just a fancy `git` command wrapper, which seems to be much of its appeal. Those who need plumbing are most certainly better off with libgit2 bindings alone - it's certainly a different difficulty level though.
